packages/core/src/factory.ts

```diff
@@ -39,93 +41,114 @@ export function factory<Resource, Input, Output, Config, LoadOpts>(
         const configObj: Config | undefined = configMerger(input);
         const inputValidations: IPrismDiagnostic[] = [];
 
-        // find the correct resource
-        let resource: Resource | undefined;
         if (components.router) {
-          try {
-            resource = components.router.route({ resources, input, config: configObj }, defaultComponents.router);
-          } catch (error) {
-            // rethrow error we if we're attempting to mock
-            if ((configObj as IPrismConfig).mock) {
-              throw error;
-            }
-            const { message, name, status } = error as ProblemJsonError;
-            // otherwise let's just stack it on the inputValidations
-            // when someone simply wants to hit an URL, don't block them
-            inputValidations.push({
-              message,
-              source: name,
-              code: status,
-              severity: DiagnosticSeverity.Warning,
-            });
-          }
-        }
+          return components.router
+            .route({ resources, input, config: configObj }, defaultComponents.router)
+            .fold(
+              error => {
+                // rethrow error we if we're attempting to mock
+                if ((configObj as IPrismConfig).mock) {
+                  return left2v(error);
+                }
 
-        // validate input
-        if (resource && components.validator && components.validator.validateInput) {
-          inputValidations.push(
-            ...(await components.validator.validateInput(
-              {
-                resource,
-                input,
-                config: configObj,
-              },
-              defaultComponents.validator,
-            )),
-          );
-        }
+                const { message, name, status } = error as ProblemJsonError;
+                // otherwise let's just stack it on the inputValidations
+                // when someone simply wants to hit an URL, don't block them
+                inputValidations.push({
+                  message,
+                  source: name,
+                  code: status,
+                  severity: DiagnosticSeverity.Warning,
+                });
 
-        // build output
-        let output: Output | undefined;
-        if (resource && components.mocker && (configObj as IPrismConfig).mock) {
-          // generate the response
-          output = components.mocker
-            .mock(
-              {
-                resource,
-                input: { validations: { input: inputValidations }, data: input },
-                config: configObj,
+                return right2v<Error, Resource | undefined>(undefined);
```

Comment:
`right2v`, whoa what a lovely interface :P I have a suggestion 
`right2lolz` because why not? Ok enough sarcazm for today

Reply:
It's actually a good point — it has been fixed in 2.x but I didn't want to deal with breaking changes right now.

Reply:
We'll get less PR from random people. Entry threshold is set high ;)